internal/upstreamldap/upstreamldap.go

```diff
@@ -103,6 +106,19 @@ type ProviderConfig struct {
 
 	// Dialer exists to enable testing. When nil, will use a default appropriate for production use.
 	Dialer LDAPDialer
+
+	// UIDAttributeParsingOverrides are mappings between an attribute name and a way to parse it as a UID when
+	// it comes out of LDAP.
+	UIDAttributeParsingOverrides []AttributeParsingOverride
+
+	// GroupNameMappingOverrides are the mappings between an attribute name and a way to parse it as a group
+	// name when it comes out of LDAP.
+	GroupAttributeParsingOverrides []AttributeParsingOverride
+}
+
+type AttributeParsingOverride struct {
+	AttributeName string
+	OverrideFunc  func(entry *ldap.Entry) (string, error)
```

Comment:
This is cool. 😎 Might it make sense for it to be a map of attribute name to OverrideFunc? That would allow the loop below to be a map lookup.

Reply:
Resolved in 05afae60c2d3dd95e805ea4a2899513493a1e31d